### PR TITLE
APIM 7797 feat: notification for failed subscription

### PR DESCRIPTION
**Pull request by mukul-tyagi08**
## Issue

https://gravitee.atlassian.net/browse/APIM-7797

## Description

Added feature to have notification for failed subscription(s) via notification configuration for API.

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/notification/ApplicationHook.java [hunk 1]
```diff
@@ -27,6 +27,7 @@ public enum ApplicationHook implements Hook {
     SUBSCRIPTION_RESUMED("Subscription Resumed", "Triggered when a Subscription is resumed.", "SUBSCRIPTION"),
     SUBSCRIPTION_REJECTED("Subscription Rejected", "Triggered when a Subscription is rejected.", "SUBSCRIPTION"),
     SUBSCRIPTION_TRANSFERRED("Subscription Transferred", "Triggered when a Subscription is transferred.", "SUBSCRIPTION"),
+    SUBSCRIPTION_FAILED("Subscription Failed", "Triggered when a Subscription is failed.", "SUBSCRIPTION"),
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/notification/ApplicationHook.java]: should be "Triggered when a subscription fails"

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/notification/ApiHook.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ public enum ApiHook implements Hook {
     SUBSCRIPTION_RESUMED("Subscription Resumed", "Triggered when a Subscription is resumed.", "SUBSCRIPTION"),
     SUBSCRIPTION_REJECTED("Subscription Rejected", "Triggered when a Subscription is rejected.", "SUBSCRIPTION"),
     SUBSCRIPTION_TRANSFERRED("Subscription Transferred", "Triggered when a Subscription is transferred.", "SUBSCRIPTION"),
+    SUBSCRIPTION_FAILED("Subscription Failed", "Triggered when a Subscription is failed.", "SUBSCRIPTION"),
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/notification/ApiHook.java]: should be "Triggered when a subscription fails"

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/notifications/portal/API.SUBSCRIPTION_FAILED.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+title: |
+  [${api.name}] Subscription Failed
+message: |
+  The subscription request for the application "${application.name}" 
+  to the plan "${plan.name}" failed.
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/notifications/portal/API.SUBSCRIPTION_FAILED.yml]: to the plan "${plan.name}" has failed.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/notifications/portal/APPLICATION.SUBSCRIPTION_FAILED.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+title: |
+  [${application.name}] Subscription Failed
+message: |
+  The subscription request to the plan "${plan.name}" 
+  of the api "${api.name}" failed.
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/notifications/portal/APPLICATION.SUBSCRIPTION_FAILED.yml]: to the plan "${plan.name}" has failed.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/subscriptionFailed.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	<body style="text-align: center;">
+		<header>
+			<#include "header.html" />
+		</header>
+		<div style="margin-top: 50px; color: #424e5a;">
+			<h3>Hi,</h3>
+			<p>The subscription to the API <code>${api.name}</code> with plan <code>${plan.name}</code> has failed.</p>
+
+			<#if subscription.reason??>
+            Message
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/src/main/resources/templates/subscriptionFailed.html:11]: why message? Can you show how it looks like in email?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionFailureCommandListenerImpl.java [hunk 6]
```diff
@@ -24,6 +24,7 @@
 import io.gravitee.rest.api.model.command.CommandTags;
 import io.gravitee.rest.api.service.SubscriptionCommandListener;
 import io.gravitee.rest.api.service.SubscriptionService;
+import io.gravitee.rest.api.service.common.ExecutionContext;
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionFailureCommandListenerImpl.java]: to remove

**Review by skorda42 — APPROVED** [review]

**Pull request merged**